### PR TITLE
More stubs for DMStandard

### PR DESCRIPTION
**Pull request by mumencoder**

### CODE REVIEW

File: DMCompiler/DMStandard/UnsortedAdditions.dm [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+/DBQuery
+	proc/_dm_db_new_query()
+	proc/_dm_db_execute()
+	proc/_dm_db_next_row()
+	proc/_dm_db_rows_affected()
+	proc/_dm_db_row_count()
+	proc/_dm_db_error_msg()
+	proc/_dm_db_columns()
+	proc/_dm_db_close()
```
Comment (wixoaGit) [inline DMCompiler/DMStandard/UnsortedAdditions.dm:161]: These are not valid types
```suggestion
```

File: DMCompiler/DMStandard/UnsortedAdditions.dm [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+/proc/rgb2num(color, space)
+/proc/spantext_char(Haystack,Needles,Start=1)
+/proc/winget(player, control_id, params)
+/proc/winexists(player, control_id)
+/proc/winclone(player, window_name, clone_name)
+/proc/winshow(player, window, show=1)
+/proc/walk_rand(Ref,Lag=0,Speed=0)
+
+/atom
+	var/filters = list()
```
Comment (wixoaGit) [inline DMCompiler/DMStandard/UnsortedAdditions.dm:23]: I worry that stubbing vars like this is going to lead to silently broken code. I'd like to have some way of flagging vars like this as unimplemented, similar to what we have for procs right now. This was brought up in #368.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Comment by ike709** [comment]
You can now flag unimplemented vars by giving them the `opendream_unimplemented` astype like `var/filters = list() as opendream_unimplemented`

File: DMCompiler/DMStandard/Types/Atoms/Movable.dm [hunk 3]
```diff
@@ -2,6 +2,8 @@
 	var/screen_loc
 
 	var/animate_movement = FORWARD_STEPS
+	var/list/locs = list()
+	var/glide_size
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Atoms/Movable.dm:6]:
```suggestion
	var/list/locs = list() as opendream_unimplemented
	var/glide_size as opendream_unimplemented
```
```suggestion
	var/list/locs = list()
	var/glide_size
```

File: DMCompiler/DMStandard/Types/Atoms/_Atom.dm [hunk 4]
```diff
@@ -37,6 +37,29 @@
...
+	var/step_y
+	var/render_source
+	var/bound_width
+	var/bound_height
+	var/mouse_drag_pointer
+	var/mouse_drop_pointer
+	var/render_target
+	var/vis_flags
+	var/vis_locs = list()
+	var/list/vis_contents = list()
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Atoms/_Atom.dm:61]:
```suggestion
	var/filters = list() as opendream_unimplemented
	var/appearance as opendream_unimplemented
	var/appearance_flags as opendream_unimplemented
	var/maptext_width as opendream_unimplemented
	var/maptext_height as opendream_unimplemented
	var/maptext_x = 32 as opendream_unimplemented
	var/maptext_y = 32 as opendream_unimplemented
	var/pixel_x as opendream_unimplemented
	var/pixel_y as opendream_unimplemented
	var/pixel_z as opendream_unimplemented
	var/pixel_w as opendream_unimplemented
	var/step_x as opendream_unimplemented
	var/step_y as opendream_unimplemented
	var/render_source as opendream_unimplemented
	var/bound_width as opendream_unimplemented
	var/bound_height as opendream_unimplemented
	var/mouse_drag_pointer as opendream_unimplemented
	var/mouse_drop_pointer as opendream_unimplemented
	var/render_target as opendream_unimplemented
	var/vis_flags as opendream_unimplemented
	var/vis_locs = list() as opendream_unimplemented
	var/list/vis_contents = list() as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Client.dm [hunk 5]
```diff
@@ -28,6 +28,21 @@
...
 	var/timezone
 
+	var/color = 0
+	var/control_freak
+	var/mouse_pointer_icon
+	var/preload_rsc = 1
+	var/fps = 0
+	var/dir = NORTH
+	var/gender = "neuter"
+	var/glide_size
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Client.dm:38]:
```suggestion
	var/color = 0 as opendream_unimplemented
	var/control_freak as opendream_unimplemented
	var/mouse_pointer_icon as opendream_unimplemented
	var/preload_rsc = 1 as opendream_unimplemented
	var/fps = 0 as opendream_unimplemented
	var/dir = NORTH as opendream_unimplemented
	var/gender = "neuter" as opendream_unimplemented
	var/glide_size as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/World.dm [hunk 6]
```diff
@@ -39,6 +39,25 @@
 	proc/New()
 	proc/Del()
 
+	var/map_cpu = 0
+	var/hub
+	var/hub_password
+	var/reachable
+	var/game_state
+	var/host
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/World.dm:47]:
```suggestion
	var/map_cpu = 0 as opendream_unimplemented
	var/hub as opendream_unimplemented
	var/hub_password as opendream_unimplemented
	var/reachable as opendream_unimplemented
	var/game_state as opendream_unimplemented
	var/host as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Sound.dm [hunk 7]
```diff
@@ -10,6 +10,10 @@
 	var/y
 	var/z
 
+	var/environment
+	var/echo
+	var/len
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Sound.dm:15]:
```suggestion
	var/environment as opendream_unimplemented
	var/echo as opendream_unimplemented
	var/len as opendream_unimplemented
```
```suggestion
	var/environment
	var/echo
	var/len
```

File: DMCompiler/DMStandard/Types/Mutable_Appearance.dm [hunk 8]
```diff
@@ -1,27 +1,8 @@
...
-	var/maptext_y = 0
-	var/maptext_height = 32
-	var/maptext_width = 32
-	
-	var/blend_mode = 0
-	var/appearance_flags = 0
-	
-	var/mouse_opacity = 1
+	var/animate_movement = 1
+	var/screen_loc
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Mutable_Appearance.dm:5]:
```suggestion
	var/animate_movement = 1 as opendream_unimplemented
	var/screen_loc as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Image.dm [hunk 9]
```diff
@@ -1,20 +1,59 @@
...
+	var/alpha = 255
+	var/appearance
+	var/appearance_flags = 0
+	var/blend_mode = 0
+	var/color = "#FFFFFF"
+	var/desc = null
+	var/gender = "neuter"
+	var/infra_luminosity = 0
+	var/invisibility = 0
+	var/list/filters = list()
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Image.dm:13]:
```suggestion
	var/alpha = 255
	var/appearance as opendream_unimplemented
	var/appearance_flags = 0 as opendream_unimplemented
	var/blend_mode = 0 as opendream_unimplemented
	var/color = "#FFFFFF"
	var/desc = null
	var/gender = "neuter" as opendream_unimplemented
	var/infra_luminosity = 0 as opendream_unimplemented
	var/invisibility = 0 as opendream_unimplemented
	var/list/filters = list() as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Image.dm [hunk 10]
```diff
@@ -1,20 +1,59 @@
...
+	var/maptext_y = 0
+	var/mouse_over_pointer = 0
+	var/mouse_drag_pointer = 0
+	var/mouse_drop_pointer = 1
+	var/mouse_drop_zone = 0
+	var/mouse_opacity = 1
+	var/name = "image"
+	var/opacity = 0
+	var/list/overlays = list()
+	var/override = 1 //TODO
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Image.dm:29]:
```suggestion
	var/luminosity = 0 as opendream_unimplemented
	var/maptext = "i" as opendream_unimplemented
	var/maptext_width = 32 as opendream_unimplemented
	var/maptext_height = 32 as opendream_unimplemented
	var/maptext_x = 0 as opendream_unimplemented
	var/maptext_y = 0 as opendream_unimplemented
	var/mouse_over_pointer = 0 as opendream_unimplemented
	var/mouse_drag_pointer = 0 as opendream_unimplemented
	var/mouse_drop_pointer = 1 as opendream_unimplemented
	var/mouse_drop_zone = 0 as opendream_unimplemented
	var/mouse_opacity = 1
	var/name = "image"
	var/opacity = 0 as opendream_unimplemented
	var/list/overlays = list()
	var/override = 1 as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Image.dm [hunk 11]
```diff
@@ -1,20 +1,59 @@
...
-	var/alpha = 255
-
-	var/blend_mode = 0
+	var/pixel_w = 0
+	var/pixel_z = 0
+	var/plane = FLOAT_PLANE
+	var/render_source
+	var/render_target
+	var/suffix
+	var/text = "i"
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Image.dm:38]:
```suggestion
	var/pixel_w = 0 as opendream_unimplemented
	var/pixel_z = 0 as opendream_unimplemented
	var/plane = FLOAT_PLANE as opendream_unimplemented
	var/render_source as opendream_unimplemented
	var/render_target as opendream_unimplemented
	var/suffix as opendream_unimplemented
	var/text = "i" as opendream_unimplemented
```

File: DMCompiler/DMStandard/Types/Image.dm [hunk 12]
```diff
@@ -1,20 +1,59 @@
...
+	var/vis_flags = 0
+
+	var/bound_width
+	var/bound_height
+	var/name
+	var/x
+	var/y
+	var/z
+	var/list/filters = list()
+	var/list/vis_contents = list()
```
Comment (ike709) [inline DMCompiler/DMStandard/Types/Image.dm:50]:
```suggestion
	var/vis_flags = 0 as opendream_unimplemented

	var/bound_width as opendream_unimplemented
	var/bound_height as opendream_unimplemented
	var/name
	var/x
	var/y
	var/z
	var/list/filters = list() as opendream_unimplemented
	var/list/vis_contents = list() as opendream_unimplemented
```

**Comment by ike709** [comment]
Should be able to just commit all of those code suggestions and merge master and this PR will be good to go.

**Comment by ike709** [comment]
conflicts

**Pull request merged**